tuist/tuist
# Fix regression on SwiftPackageManager packages defining file resources with copy rule

Resolves: https://github.com/tuist/tuist/issues/4796 and regression mentioned in https://github.com/tuist/tuist/pull/4733

### Short description 📝

This PR tries to fix a regression on SPM projects where resources are included with the `.copy` rule. It seems that the previous implementation failed for the following "type" of resources:

* Opaque directories that should be handled like files (e.g. xcdatamodel)
* Single files which are not directories 

In order to not break unit tests it is now required for SPM generation related unit tests that files and directories declared as resources need to be really present on the file system during the test. Otherwise `AbsolutePath.isDirectory` and `AbsolutePath.isFile` will not work correctly and report fails positive. 

I also changed the `app_with_spm_dependencies` fixture project to include the two cases mentioned above. Unfortunately they cannot be checked at build time. Some need to run the project to verify the files are accessible. Or to take a look at the `Copy Bundle resources` manually. 

### How to test the changes locally 🧐

Try to run `tuist fetch` and `tuist generate` for the following repositories

* https://github.com/salqueng/tuist-with-stream-chat-swift-bug (opaque files copy case)
* https://github.com/Arafo/tuist-spm-resources-copy-issue (single file copy case)
* https://github.com/alexanderwe/tuist-spm-resources-copy-issue (directory copy case)
* Generation and build of these projects should not fail

### Checklist ✅
